transaction.go

```diff
@@ -23,40 +24,86 @@ type TransactionsRequest struct {
 	To                time.Time
 	Offset            int
 	Limit             int
+	Client            *Client
+}
+
+type TransactionsIterator struct {
+	request  *TransactionsRequest
+	response TransactionsResponse
+	hasRun   bool
 }
 
 type TransactionsResponse struct {
 	Errors  []map[string]string `json:"errors"`
 	Results []Transaction       `json:"results"`
-	Pages   map[string]string   `json:"pages"`
+	Pages   Pages               `json:"pages"`
 }
 
-func (c *Client) GetTransactions(transactionsReq TransactionsRequest) ([]Transaction, error) {
+func (t *TransactionsRequest) get(params *url.Values) (TransactionsResponse, error) {
 	var err error
 	var req *http.Request
-	var transactions []Transaction
+	var response TransactionsResponse
+
+	var url *url.URL
+	if url, err = url.Parse(fmt.Sprintf("%s/%s", ApiBase, "transactions/")); err != nil {
+		return response, err
+	}
+
+	if params != nil {
+		url.RawQuery = params.Encode()
+	}
 
-	if req, err = http.NewRequest("GET", fmt.Sprintf("%s/%s", ApiBase, "transactions/"), nil); err != nil {
-		return transactions, err
+	if req, err = http.NewRequest("GET", url.String(), nil); err != nil {
+		return response, err
 	}
 	var resp *http.Response
 
-	if resp, err = c.Do(req); err != nil {
-		return transactions, err
+	if resp, err = t.Client.do(req); err != nil {
+		return response, err
 	}
 
-	var response TransactionsResponse
 	if err = json.NewDecoder(resp.Body).Decode(&response); err != nil {
-		return transactions, err
+		return response, err
 	}
+	return response, nil
+}
 
-	return transactions, nil
+func (r *TransactionsRequest) Iterator() TransactionsIterator {
+	return TransactionsIterator{
+		request: r,
+	}
 }
 
-func (c *Client) Do(req *http.Request) (*http.Response, error) {
-	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", c.accessToken))
-	if c.clientVersion != "" {
-		req.Header.Set("Client-Version-Id", c.clientVersion)
+func (t *TransactionsIterator) Next() error {
+	var err error
+	if t.response, err = t.request.get(t.response.Pages.Next); err != nil {
+		return err
 	}
-	return c.httpClient.Do(req)
+	t.hasRun = true
```

Comment:
i don't really like having to use an extra variable...any ideas of how to handle the situation where the `Next()` hasn't been called yet, so a `HasNext()` should be true?